### PR TITLE
Prepare SPV maintainer for redemptions

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Refs: https://github.com/keep-network/keep-core/issues/3615

The current code of SPV maintainer is strongly tied to deposit sweeps. In order to implement redemption proofs in an efficient and clean way, we need to make them more generic and extract common parts that can be reused by different proof types in order to avoid duplication. Here we aim to achieve this. I recommend reviewing commit by commit to understand all changes being part of this refactoring.

### CODE REVIEW

File: pkg/maintainer/spv/config.go [hunk 1]
```diff
@@ -34,27 +35,27 @@ type Config struct {
...
-	// the yet-not-proven sweep.
+	// maintainer establishes the list of proposals of given type, based on the
+	// proposal submission events from the wallet coordinator contract, it needs
+	// to check Bitcoin transactions executed by the wallet. Then, it tries to
+	// find the transactions matching the given proposal type. For example, if set
+	// to `20`, only the latest twenty transactions will be returned. This
+	// value must not be too high so that the transaction lookup is efficient.
+	// At the same time, this value can not be too low to make sure the
+	// performed proposal's transaction can be found in case the wallet decided
+	//to execute some other Bitcoin transaction after the yet-not-proven transaction.
```
Comment (tomaszslabon) [inline pkg/maintainer/spv/config.go:58]:
```suggestion
	// to execute some other Bitcoin transaction after the yet-not-proven 
	// transaction.
```

File: pkg/maintainer/spv/spv.go [hunk 2]
```diff
@@ -95,26 +112,58 @@ func (sm *spvMaintainer) maintainSpv(ctx context.Context) error {
...
+			"failed to get unproven transactions: [%v]",
 			err,
 		)
 	}
 
 	logger.Infof(
-		"found [%d] unproven deposit sweep transaction(s)",
-		len(depositSweepTransactions),
+		"found [%d] unproven transaction(s)",
+		len(transactions),
```
Comment (tomaszslabon) [inline pkg/maintainer/spv/spv.go:158]:
```suggestion
		logger.Infof("found [%d] unproven transaction(s)", len(transactions))
```

File: pkg/maintainer/spv/spv.go [hunk 3]
```diff
@@ -95,26 +112,58 @@ func (sm *spvMaintainer) maintainSpv(ctx context.Context) error {
...
+		sm.config.HistoryDepth,
+		sm.config.TransactionLimit,
+		sm.btcChain,
+		sm.spvChain,
+	)
 	if err != nil {
 		return fmt.Errorf(
-			"failed to get unproven deposit sweep transactions: [%v]",
+			"failed to get unproven transactions: [%v]",
 			err,
```
Comment (tomaszslabon) [inline pkg/maintainer/spv/spv.go:152]: The line is probably shorter than 80 chars now.
```suggestion
			return fmt.Errorf("failed to get unproven transactions: [%v]", err)
```

File: pkg/maintainer/spv/config.go (hunk 1 above)
Comment (lukasz-zimnoch) [reply pkg/maintainer/spv/config.go:58]: Addressed in 1953e4a0

File: pkg/maintainer/spv/spv.go (hunk 2 above)
Comment (lukasz-zimnoch) [reply pkg/maintainer/spv/spv.go:158]: Addressed in 1953e4a0

File: pkg/maintainer/spv/spv.go (hunk 3 above)
Comment (lukasz-zimnoch) [reply pkg/maintainer/spv/spv.go:152]: Addressed in 1953e4a0

**Pull request merged**